### PR TITLE
network and participants, dapr

### PR DESCRIPTION
**Pull request by siredmar**
This PR 

* handles a new networks and participants structure
[x] CRD networks.network.edgefarm.io now handles public jetstreams. Creates secrets for the applications with the jetstream creds.
[x] CRD participants.network.edgefarm.io introduced. It patches the corresponding network.
[x] Added some docs

* create the requirements for edgefarm.applications components to use dapr with edgefarm.network
[x] new secret format: one secret for each component containing the credentials to each network the component is participating in.
[x] talks to anck-credentials v0.4.0
[x] adds one dapr secret per edgefarm.applications component containing all network pubsub jetstream configurations the component is participating in.

### CODE REVIEW

File: controllers/network/jetstreams.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+	_, err = mgr.LoadOrNewStreamFromDefault(streamConfig.Name, *opts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Delete deletes a jetstream stream
+func (j *Jetstream) Delete(names []string) error {
```
Comment (KairaMiu) [inline controllers/network/jetstreams.go:73]: Rename of stream leads to a new jetstream, but the old one is never deleted:
- Create network `networkA` with jetstream `stream1`
- Jetstreams: `stream1`
- Update ressource network `networkA`: rename jetstream `stream1` to `stream2` and apply
- Jetstreams: `stream1`, `stream2`
- Delete ressource network `networkA`
- Jetstreams: `stream1`

**Review by KairaMiu — CHANGES_REQUESTED** [review]

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-  - files:
-      - controller_manager_config.yaml
-    name: manager-config
+- files:
+  - controller_manager_config.yaml
+  name: manager-config
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
-  - name: controller
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply config/manager/kustomization.yaml:14]: Fixed in review PR

File: config/samples/network_v1alpha1_network-account-only-sample.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: my-account-only-sample
 spec:
   accountname: my-account-only
-  usernames:
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply config/samples/network_v1alpha1_network-account-only-sample.yaml:7]: Fixed in review PR.

File: docs/images/dapr.puml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+@startuml
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply docs/images/dapr.puml:1]: Fixed in review PR

File: docs/images/global-streams.puml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+@startuml
+
+package "global-network" as global_network {
+    [configure streams] as configure_streams
+    database "global nats" as global_nats
+    configure_streams --> global_nats
+}
+
+folder "global_config" {
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply docs/images/global-streams.puml:9]: Fixed in review PR

File: devspace.yaml [hunk 6]
```diff
@@ -36,85 +36,67 @@ commands:
     command: |-
       devspace update dependencies
 
-dev:
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply devspace.yaml:39]: Fixed in review PR

File: pkg/dapr/dapr.go [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+// WithCreds sets the credentials for the DaprConfig
+func WithCreds(jwt string, nkey string) Option {
+	return func(c *Config) {
+		c.Spec.SpecMetadata = append(c.Spec.SpecMetadata, SpecMetadata{
+			Name:  "jwt",
+			Value: jwt,
+		})
+		c.Spec.SpecMetadata = append(c.Spec.SpecMetadata, SpecMetadata{
+			Name:  "seedkey",
```
Comment (KairaMiu) [inline pkg/dapr/dapr.go:46]: Shall be `seedKey`

**Review by KairaMiu — CHANGES_REQUESTED** [review]

File: pkg/dapr/dapr.go (hunk 7 above)
Comment (KairaMiu) [reply pkg/dapr/dapr.go:46]: Fixed in review PR

**Review by KairaMiu — CHANGES_REQUESTED** [review]
General: Secret entries are deleted, when participant is removed.  Secret itself is not deleted, when no participant left. Is this intended?

File: controllers/network/dapr.go [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+		_, err = createSecret(daprSecretName, secret.Namespace, &items)
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func removeParticipantFromDaprSecret(component, network, namespace string) error {
```
[the remark this answers is not in the document]
Comment (KairaMiu) [reply controllers/network/dapr.go:45]: This is fixed in review PR.

[the remark this answers is not in the document]
Comment (KairaMiu) [reply controllers/network/dapr.go:45]: Fixed in review PR.

File: main.go [hunk 9]
```diff
@@ -75,13 +75,27 @@ func main() {
...
 		os.Exit(1)
 	}
+	if err = (&networkcontrollers.ParticipantsReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "Participants")
+		os.Exit(1)
+	}
+	if err = (&networkcontrollers.ParticipantsReconciler{
```
[the remark this answers is not in the document]
Comment (siredmar) [reply main.go:92]: this is an error

File: config/manager/kustomization.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (siredmar) [reply config/manager/kustomization.yaml:14]: cool!

**Review by KairaMiu — APPROVED** [review]

**Pull request merged**